packages/block-editor/src/components/block-list/use-multi-selection.js

```diff
@@ -30,35 +35,32 @@ function getDeepestNode( node, type ) {
 	return node;
 }
 
-export default function useMultiSelection( { ref, rootClientId } ) {
-	function selector( select ) {
-		const {
-			getBlockOrder,
-			isSelectionEnabled,
-			isMultiSelecting,
-			getMultiSelectedBlockClientIds,
-			hasMultiSelection,
-			getBlockParents,
-		} = select( 'core/block-editor' );
-
-		return {
-			blockClientIds: getBlockOrder( rootClientId ),
-			isSelectionEnabled: isSelectionEnabled(),
-			isMultiSelecting: isMultiSelecting(),
-			multiSelectedBlockClientIds: getMultiSelectedBlockClientIds(),
-			hasMultiSelection: hasMultiSelection(),
-			getBlockParents,
-		};
-	}
+function selector( select ) {
```

Comment:
Aside: While the separate function declaration still catches me off guard, one thing which is nice here is demonstrating how, if you are able to move it outside the component declaration altogether, there's a safer guarantee that you don't need to provide any dependencies to the `useSelect` call (since there's no longer the component scope from which to inherit dependencies).

Reply:
I agree. It's maybe a bit strange to put it outside the component, but why not if you can prevent creating an additional function on every render.